packages/astro/src/core/config/config.ts

```diff
@@ -248,5 +238,25 @@ export function createDefaultDevConfig(
 	userConfig: AstroUserConfig = {},
 	root: string = process.cwd()
 ) {
-	return resolveConfig(userConfig, root, undefined, 'dev');
+	return resolveConfigLegacy(userConfig, root, undefined, 'dev');
+}
+
+interface ResolveConfigResult {
+	userConfig: AstroUserConfig;
+	astroConfig: AstroConfig;
+}
+
+// NOTE: This is the new API to load an Astro config. Use this instead of the other functions!
+export async function resolveConfig(
+	inlineConfig: AstroInlineConfig,
+	command: string,
+	fsMod = fs
+): Promise<ResolveConfigResult> {
```

Comment:
I think it's worth documenting :)

Reply:
Definitely after I get get all the refactors done 😄 I'm not completely sure this is the final shape yet so sort-of an interim code.